java-squid/src/test/files/sym/references/MethodCall.java

```diff
@@ -181,3 +181,29 @@ void varargs_usage_2() {
     varargs2("", "", "");
   }
 }
+
+class GenericClass {
+
+  class NestedGenericClass<T> {
+    private NestedGenericClass(T argument) {
+    }
+
+    private void genericMethod(T argument) {
+    }
+  }
+
+  class ComplexNestedGenericClass<T extends java.util.Collection<Object>> {
+    private ComplexNestedGenericClass(T argument) {
+    }
+
+    private void complexGenericMethod(T argument) {
+    }
+  }
+
+  public void test() {
+    NestedGenericClass<List<Object>> nestedGenericClass = new NestedGenericClass<List<Object>>(new java.util.ArrayList<Object>());
+    nestedGenericClass.genericMethod(new java.util.LinkedList<Object>());
+    ComplexNestedGenericClass<List<Object>> complexNestedGenericClass = new ComplexNestedGenericClass<List<Object>>(new java.util.ArrayList<Object>());
+    complexNestedGenericClass.complexGenericMethod(new java.util.LinkedList<Object>());
+  }
```

Comment:
Not sure that we are testing all the possible situations there.
On my side I also tested with the following cases, without encountering any issues:
```
    ComplexNestedGenericClass v1 = new ComplexNestedGenericClass(new java.util.ArrayList<Object>());
    ComplexNestedGenericClass<U> v2 = new ComplexNestedGenericClass<U>(myU);
    ComplexNestedGenericClass<? extends List<Object>> v3 = new ComplexNestedGenericClass<List<Object>>(new java.util.ArrayList<Object>());
```